src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -164,41 +168,76 @@ def _get_shape(self, evt):
                     return shape
         return None
 
+    def _create_new_shape(self):
+        """Create a new shape."""
```

Comment:
This is a odd function which create a shape *as side-effect*. At least, I'd recommend to return the shape that was just returned. 
Maybe then the caller could do `self._selected_shape = self._create_new_shape()`, which would make things a little less "magical" when reading the code.